test/cli_snapshotter_test.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/alibaba/pouch/test/environment"
+
+	"github.com/go-check/check"
+)
+
+// PouchSnapshotterSuite is the test suite for choose snapshotter feature.
+type PouchSnapshotterSuite struct{}
+
+func init() {
+	check.Suite(&PouchSnapshotterSuite{})
+}
+
+// SetUpSuite does common setup in the beginning of each test suite.
+func (suite *PouchSnapshotterSuite) SetUpSuite(c *check.C) {
+	SkipIfFalse(c, environment.IsLinux)
+	environment.PruneAllContainers(apiClient)
+}
+
+// TearDownTest does cleanup work in the end of each test.
+func (suite *PouchSnapshotterSuite) TearDownTest(c *check.C) {
+}
+
+// TestNotSetSnapshotter tests default snapshotter to run pouchd
+func (suite *PouchSnapshotterSuite) TestNotSetSnapshotter(c *check.C) {
+	dcfg, err := StartDefaultDaemon()
+	c.Assert(err, check.IsNil)
+
+	defer dcfg.KillDaemon()
+
+	result := RunWithSpecifiedDaemon(dcfg, "pull", busyboxImage)
+	c.Assert(result.ExitCode, check.Equals, 0)
+	// clean busybox image
+	defer RunWithSpecifiedDaemon(dcfg, "rmi", busyboxImage)
+
+	// we can get snapshots dir from overlayfs dirs
+	_, err = checkSnapshotsDir(dcfg.HomeDir, "overlayfs")
+	c.Assert(err, check.IsNil)
+
+}
+
+// TestSetDefaultSnapshotter tests set default snapshotter driver to run pouchd
+func (suite *PouchSnapshotterSuite) TestSetDefaultSnapshotter(c *check.C) {
+	dcfg, err := StartDefaultDaemon("--snapshotter", "overlayfs")
+	c.Assert(err, check.IsNil)
+
+	defer dcfg.KillDaemon()
+
+	result := RunWithSpecifiedDaemon(dcfg, "pull", busyboxImage)
+	c.Assert(result.ExitCode, check.Equals, 0)
+	// clean busybox image
+	defer RunWithSpecifiedDaemon(dcfg, "rmi", busyboxImage)
+
+	// we can get snapshots dir from overlayfs dirs
+	_, err = checkSnapshotsDir(dcfg.HomeDir, "overlayfs")
+	c.Assert(err, check.IsNil)
+}
+
+// TestOldSnapshotterNotClean tests old snapshotter driver not clean and then set a new one
+func (suite *PouchSnapshotterSuite) TestOldSnapshotterNotClean(c *check.C) {
+	dcfg, err := StartDefaultDaemon("--snapshotter", "overlayfs")
+	c.Assert(err, check.IsNil)
+
+	result := RunWithSpecifiedDaemon(dcfg, "pull", busyboxImage)
+	c.Assert(result.ExitCode, check.Equals, 0)
+	dcfg.KillDaemon()
+	time.Sleep(10 * time.Second)
+
+	_, err = StartDefaultDaemon("--snapshotter", "btrfs")
```

Comment:
I don't think all the machines can support `btrfs`. The `btrfs` should be supported by `btrfs filesystem`. I think we can check for precondition before run test case. If the precondition doesn't valid, we should skip the case. WDYT?